app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -35,56 +37,49 @@ class Api::V2::RepositoriesController < Api::V2::ApiController
     param :content_type, String, :desc => N_("limit to only repositories of this time")
     param :name, String, :desc => N_("name of the repository"), :required => false
+    param :available_for, String, :desc => N_("interpret specified object to return only Repositories that can be associated with specified object.  Only 'content_view' is supported."),
+          :required => false
     param_group :search, Api::V2::ApiController
     # rubocop:disable Metrics/MethodLength
     def index
-      options = sort_params
-      options[:load_records?] = true
-
-      repositories = Repository.where(:product_id => Product.readable.where(:organization_id => @organization.id))
-      repositories = repositories.where(:product_id => @product.id) if @product
-
-      if params[:content_view_id] && params[:environment_id]
-        version = ContentViewVersion.in_environment(params[:environment_id]).where(:content_view_id => params[:content_view_id])
-        repositories = repositories.where(:content_view_version_id => version)
-      elsif params[:content_view_id]
-        repositories = repositories
-                         .joins(:content_view_repositories)
-                         .where("#{ContentViewRepository.table_name}.content_view_id" => params[:content_view_id])
-      end
+      options = {:includes => [:gpg_key, :product, :environment]}
+      respond(:collection => scoped_search(index_relation.uniq, :name, :desc, options))
+    end
 
-      repositories = repositories.where(:content_type => params[:content_type]) if params[:content_type]
-      repositories = repositories.where(:name => params[:name]) if params[:name]
-      repositories = repositories.joins(:errata).where("#{Erratum.table_name}.uuid" => params[:erratum_id]) if params[:erratum_id]
+    def index_relation
+      query = Repository.where(:product_id => Product.readable.where(:organization_id => @organization.id))
+      query = query.where(:product_id => @product.id) if @product
+      query = query.where(:content_type => params[:content_type]) if params[:content_type]
+      query = query.where(:name => params[:name]) if params[:name]
+      query = query.joins(:errata).where("#{Erratum.table_name}.uuid" => params[:erratum_id]) if params[:erratum_id]
 
-      if params[:content_view_version_id]
-        repositories = repositories.where(:content_view_version_id => params[:content_view_version_id])
+      if params[:content_view_id]
+        query = filter_by_content_view(query, params[:content_view_id], params[:environment_id], params[:available_for] == 'content_view')
+      end
 
-        if params[:library]
-          repositories = Repository.where(:id => repositories.map(&:library_instance_id))
-        end
+      if params[:content_view_version_id]
+        query = query.where(:content_view_version_id => params[:content_view_version_id])
+        query = Repository.where(:id => query.map(&:library_instance_id)) if params[:library]
       end
```

Comment:
Seems like if I provide a content_view_version_id and a content_view_id then I wouldn't need to perform the `filter_by_content_view` method?